tests/components/bosch_alarm/test_services.py

```diff
@@ -0,0 +1,168 @@
+from homeassistant.util import dt as dt_util
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(autouse=True)
+async def platforms() -> AsyncGenerator[None]:
+    """Return the platforms to be loaded for this test."""
+    with patch("homeassistant.components.bosch_alarm.PLATFORMS", []):
+        yield
+
+
+async def test_set_date_time_service(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls succeed if the service call is valid."""
+    await setup_integration(hass, mock_config_entry)
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_DATE_TIME,
+        {
+            ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
+            ATTR_DATETIME: dt_util.now(),
+        },
+        blocking=True,
+    )
+    mock_panel.set_panel_date.assert_called_once()
+
+
+async def test_set_date_time_service_fails_bad_entity(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the service call is done for an incorrect entity."""
+    await setup_integration(hass, mock_config_entry)
+    with pytest.raises(
+        ServiceValidationError,
+        match='Integration "bad-config_id" not found in registry',
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SET_DATE_TIME,
+            {
+                ATTR_CONFIG_ENTRY_ID: "bad-config_id",
+                ATTR_DATETIME: dt_util.now(),
+            },
+            blocking=True,
+        )
+
+
+async def test_set_date_time_service_fails_bad_params(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the service call is done with incorrect params."""
+    await setup_integration(hass, mock_config_entry)
+    with pytest.raises(
+        vol.MultipleInvalid,
+        match=r"Invalid datetime specified:  for dictionary value @ data\['datetime'\]",
+    ):
+        await hass.services.async_call(
+            DOMAIN,
+            SERVICE_SET_DATE_TIME,
+            {
+                ATTR_CONFIG_ENTRY_ID: mock_config_entry.entry_id,
+                ATTR_DATETIME: "",
+            },
+            blocking=True,
+        )
+
+
+async def test_set_date_time_service_fails_bad_year(
+    hass: HomeAssistant,
+    mock_panel: AsyncMock,
+    area: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that the service calls fail if the panel fails the service call."""
+    await setup_integration(hass, mock_config_entry)
+    mock_panel.set_panel_date.side_effect = ValueError()
+    with pytest.raises(
+        ServiceValidationError,
+        match="Bosch alarm panels only support years between 2010 and 2037",
```

Comment:
Why don't you validate this using the validation schema?

Reply:
that is a very good point